### ISSUE TITLE
Investigate possibly broken 8.10.1

### ISSUE THREAD

**Issue by ljacomet** [gradle/gradle#30472]
### Current Behavior

As seen in #30462, Gradle fails to update the wrapper on `release` to 8.10.1 with:

```
> Cannot call Task.usesService(Provider) on task ':build-logic:integration-testing-kotlin:compileKotlin' after task has started execution
```

### Expected Behavior

This should not happen

### Steps to Reproduce

Take the code from #30462 and:

1. `./gradlew -g test-home sanCh`
2. `./gradlew -g test-home prepareKotlinBuildScriptModel :build-logic:binary-compatibility:compileTestKotlin`

Can reproduce by doing 2 before 1 as well, make sure you get a CC miss though (add a newline to `settings.gradle.kts` is enough) and stop daemon between runs: `./gradlew -g test-home --stop`

### Gradle version

8.10.1

### Build scan URL (optional)

https://ge.gradle.org/s/acaycvsfdqn4c

**Comment by jasperkamerling** [comment]
When upgrading to 8.10.1 I regularly get a similar but different error, see: https://gist.github.com/jasperkamerling/0e5d582b3924d383d58c7a96fdb5cb91
If it helps i can provide the projects build scripts.

**Comment by paul-middelkoop-sympower** [comment]
We encounter the same issue as @jasperkamerling in many of our build.

**Comment by 3flex** [comment]
> When upgrading to 8.10.1 I regularly get a similar but different error, see: https://gist.github.com/jasperkamerling/0e5d582b3924d383d58c7a96fdb5cb91

Seeing this one too in detekt builds since 8.10.1. Should this be raised as a separate issue or is it suspected to have the same root cause as `Cannot call Task.usesService(Provider) on task`?

It's intermittent so I don't have a reproducer.

**Comment by JWood48** [comment]
I am also seeing this in a large multi includeBuild project when importing to eclipse with buildship - happens every time

**Comment by alyssa-nash** [comment]
I managed to have a temp workaround by adding `--no-daemon` to my commands.

**Comment by GingerGeek** [comment]
It seems this occurs on daemon reuse. I can workaround by using `gradle --stop`, `./gradlew --stop` or `.\gradlew.bat --stop` when it happens.

**Comment by ljacomet** [comment]
The root cause has been identified and is indeed related to the Kotlin plugins keeping stale state in the daemon. We are [validating a fix](https://github.com/gradle/gradle/pull/30488).

If you are impacted, either:
* Revert to Gradle 8.10 until we release 8.10.2
* Run your builds with `--no-daemon`, which will make sure the daemon expires at the end of the build

**Comment by 6hundreds** [comment]
@3flex 
Yes, it will

**Comment by IGR2014** [comment]
Same problem, same workaround helps

**Comment by ljacomet** [comment]
Hello @LarsSven, @jasperkamerling, @paul-middelkoop-sympower, @3flex, @JWood48, @andriperalt, @GingerGeek  and @IGR2014,

Could you try version `8.10.2-milestone-1` and report the results?

`./gradlew wrapper --gradle-version 8.10.2-milestone-1`

On our side we are [updating Gradle's own wrapper to that version](https://github.com/gradle/gradle/pull/30640) on the `release` branch to confirm the fix.

**Comment by jasperkamerling** [comment]
@ljacomet With version `8.10.2-milestone-1` I'm not able to reproduce the issue.
Thanks for the fix, looking forward to the release of `8.10.2` :)

**Comment by bamboo** [comment]
Fixed via #30523

**Comment by bric3** [comment]
For me as well this looks fixed with gradle-8.10.2-milestone-1

https://github.com/bric3/excalidraw-jetbrains-plugin/pull/340
